portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -96,7 +100,22 @@ public void storeMarker(Marker marker) {
     markerEntity.setProperty(ENTITY_PROPERTY_KEY_5, marker.getTime());
     markerEntity.setProperty(ENTITY_PROPERTY_KEY_6, marker.getAddress());
     markerEntity.setProperty(ENTITY_PROPERTY_KEY_7, marker.getDescription());
-
     datastore.put(markerEntity);
+    }
+
+  public Location getUserLocation(HttpServletRequest request){
+    String locationStr = request.getParameter("location");
+    String[] locationArrStr = locationStr.split(",", 2);
+    double lat = Double.parseDouble(locationArrStr[0]);
+    double lng = Double.parseDouble(locationArrStr[1]);
+    Location userLocation = new Location(lat, lng);
+    return userLocation;
+  }
+
+  /* Returns distance between user location and a report (in meters).*/
+  public double haversineDistance(Location location, double markerLat, double markerLng) {
+    double distance = org.apache.lucene.util.SloppyMath.haversinMeters(location.getLatitude(), location.getLongitude(), markerLat, markerLng);
```

Comment:
do an `import static org.apache.lucene.util.SloppyMath.haversinMeters` at the very top of the file. Then this line becomes
`double distance = haversinMeters(...)`